src/dotnet/commands/dotnet-cache/CacheCommand.cs

```diff
@@ -0,0 +1,126 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.DotNet.Tools.MSBuild;
+using Microsoft.DotNet.Tools.Restore;
+using Microsoft.DotNet.Cli.CommandLine;
+
+namespace Microsoft.DotNet.Tools.Cache
+{
+    public partial class CacheCommand
+    {
+        public string ProjectPath { get; set; }
+        public string Framework { get; set; }
+        public string Runtime { get; set; }
+        public string OutputPath { get; set; }
+        public string FxVersion { get; set; }
+        public string WorkingDir { get; set; }
+        public string Verbosity { get; set; }
+        private bool PreserveWorkingDir { get; set; }
+
+        
+        public List<string> ExtraMSBuildArguments { get; set; }
+
+        private CacheCommand()
+        {
+        }
+
+        private void PrepareForExecute(CommandArgument argument)
+        {
+            string projectPath = string.Empty;
```

Comment:
Based upon the usage, I would suggest renaming this to something like "TargetProjectPath".